Resolve Battleform speeds

MrVauxs opened the pull request:
Implements #20571, resolving the speed value to its intended type.

MrVauxs committed:
Resolve Battleform speeds

CarlosFdez commented:
An issue was revealed and I had to draft on his behalf since he's not a contributor. I'll undraft once he feels its ready.

MrVauxs committed:
Add check for positive speedOverride in prepareSpeeds

MrVauxs committed:
Make the fallback value null instead of adding more logic

Change the default resolve to null so explicit "0" movement speeds are allowed.
